src/parser/grammar.mly

```diff
@@ -188,25 +197,20 @@ type_:
   }
 
 atype_term:
-| c = STRING { mkC (cstring.Util.CData.cin c) }
-| c = constant { Const (fix_church c) }
+| c = STRING { TCData (cstring.Util.CData.cin c) }
+| c = constant { TConst (fix_church c) }
 | LPAREN; t = type_term; RPAREN { t }
+| LPAREN; t = anonymous_pred; RPAREN { t }
```

Comment:
maybe: the `LPAREN; t = type_term; RPAREN` rule is a valid derivation if `t` is not of the form `... -> prop`.
In this case we should use the syntax for anonymous predicates